src/services/chains/solana/transactionService.ts

```diff
@@ -0,0 +1,77 @@
+import { logger } from '../../../utils/logger';
+import SolanaWeb3, { ParsedInstruction } from '@solana/web3.js';
+import {
+  NetworkTransactionInfo,
+  TransactionDetailInput,
+  validateTransactionWithInputData,
+} from '../index';
+import { normalizeAmount } from '../../../utils/utils';
+import {
+  i18n,
+  translationErrorMessagesKeys,
+} from '../../../utils/errorMessages';
+
+const SOLANA_NODE_RPC_URL = process.env.SOLANA_NODE_RPC_URL as string;
+const solana = new SolanaWeb3.Connection(SOLANA_NODE_RPC_URL);
+
+export const getSolanaTransactionDetailForNormalTransfer = async (params: {
+  txHash: string;
+}): Promise<NetworkTransactionInfo | null> => {
+  const TRANSFER_INSTRUCTION_TYPE = 'transfer';
+  const SOL_CURRENCY_TYPE = 'SOL';
+
+  try {
+    const result = await solana.getParsedTransaction(params.txHash);
+    const data = result?.transaction?.message?.instructions?.find(
+      instruction =>
+        (instruction as ParsedInstruction)?.parsed?.type ===
+        TRANSFER_INSTRUCTION_TYPE,
```

Comment:
Let's filter based on source and destination as well. I am not sure single transfer exists in a Solana transaction

Reply:
Good idea, done